### PR TITLE
Read sample_indexes in python, this closes #49

### LINKED ISSUE #49: Access  sample_indexes from python

### PR DESCRIPTION
**Pull request by kjellkongsvik**

### CODE REVIEW

File: python/segyio/_segyio.c [hunk 1]
```diff
@@ -620,15 +620,42 @@ static Py_buffer check_and_get_buffer(PyObject *object, const char *name, unsign
     return buffer;
 }
 
+static Py_buffer check_and_get_double_buffer(PyObject *object, const char *name, unsigned int expected) {
```
Comment (kjellkongsvik) [inline python/segyio/_segyio.c]: Should we rather parametrise  check_and_get_buffer?

File: python/segyio/segy.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def __init__(self, filename, mode, iline=189, xline=193, t0=1111.0):
         # property value holders
```
Comment (jepebe) [inline python/segyio/segy.py]: `t0` should be fixed/removed in arguments

File: python/segyio/segy.py [hunk 3]
```diff
@@ -536,6 +537,10 @@ def writefn(t0, length, step, val):
     def xline(self, value):
         self.xline[:] = value
 
+    @property
+    def sample_indexes(self):
+        return self._sample_indexes
+
```
Comment (jepebe) [inline python/segyio/segy.py]: The used `t0` and `dt` should be available.

File: python/segyio/_segyio.c [hunk 4]
```diff
@@ -620,15 +620,42 @@ static Py_buffer check_and_get_buffer(PyObject *object, const char *name, unsign
...
+    PyObject_GetBuffer(object, &buffer, PyBUF_FORMAT | PyBUF_C_CONTIGUOUS | PyBUF_WRITEABLE);
+
+    if (strcmp(buffer.format, "d") != 0) {
+        PyErr_Format(PyExc_TypeError, "The destination for %s is not a buffer object of type 'double'", name);
+        PyBuffer_Release(&buffer);
+        return buffer;
+    }
+
+    size_t buffer_len = buffer.len;
+    if (buffer_len < expected * sizeof(unsigned int)) {
```
Comment (jokva) [inline python/segyio/_segyio.c]: sizeof here should be `double`.

File: python/segyio/_segyio.c [hunk 5]
```diff
@@ -657,6 +684,21 @@ static PyObject *py_init_line_indices(PyObject *self, PyObject *args) {
         return NULL;
     }
 
+    unsigned int sample_count;
+    PyArg_Parse(PyDict_GetItemString(metrics, "sample_count"), "I", &sample_count);
+    Py_buffer sample_indexes_buffer = check_and_get_double_buffer(sample_indexes_out, "sample_indexes", sample_count);
+
+    if (PyErr_Occurred()) {
```
Comment (jokva) [inline python/segyio/_segyio.c]: This leaks the iline and xline buffers.

**Comment by jokva** [comment]
I'd like to add that this or my own PR (ready some time tomorrow) *will* be in conflict in the `line_indices` function.

File: python/segyio/segy.py (hunk 2 above)
Comment (kjellkongsvik) [inline python/segyio/segy.py]: meaning?

Comment (jepebe) [inline python/segyio/segy.py]: Default `t0=1111.0` ?

File: python/segyio/_segyio.c (hunk 4 above)
Comment (kjellkongsvik) [inline python/segyio/_segyio.c]: of course

File: python/segyio/segy.py (hunk 2 above)
Comment (kjellkongsvik) [inline python/segyio/segy.py]: sure, default to t0=0

**Pull request closed without merging**